interfaces/apparmor/template.go

```diff
@@ -232,6 +232,8 @@ var templateCommon = `
   # LP: #1546825 for details). comm is a subset of cmdline, so allow it too.
   owner @{PROC}/@{pid}/cmdline r,
   owner @{PROC}/@{pid}/comm r,
+  # java attempts to read this on start-up
+  /sys/fs/cgroup/system.slice/snap.@{SNAP_INSTANCE_NAME}{,.*}/memory.max r,
```

Comment:
there is a bunch of rules related to memory controller some 50 lines down, perhaps you could move this rule closer to the existing ones

and yes, we allowed to read limit with cgroup v1 through this snippet:
```
  /sys/fs/cgroup/memory/{,**/}snap.@{SNAP_INSTANCE_NAME}{,.*}/memory.limit_in_bytes r,
```
so I don't see any reason why we shouldn't allow the same with v2

Reply:
> there is a bunch of rules related to memory controller some 50 lines down, perhaps you could move this rule closer to the existing ones

That's where it was placed initially but this cluster of rules relates to Java specifically so I moved it here in the end. I'll move it down :+1: